### PR TITLE
[wasm][debugger] Fix debugger after ts changes

### PR DESCRIPTION
**Pull request by thaystg**
Debugger was broken after https://github.com/dotnet/runtime/pull/59392

- The function name mono_wasm_fire_debugger_agent_message doesn't have the _ anymore
- Reverting usage of `local_eval`, using `local_eval` will make eval run on global context, but we need `proxy` variable that is on local context.

### CODE REVIEW

File: src/mono/wasm/runtime/src/mono/debug.ts [hunk 1]
```diff
@@ -159,8 +159,7 @@ export function mono_wasm_call_function_on(request: CallRequest) {
     const fn_args = request.arguments != undefined ? request.arguments.map(a => JSON.stringify(a.value)) : [];
     const fn_eval_str = `var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]);`;
 
-    const local_eval = eval; // https://rollupjs.org/guide/en/#avoiding-eval
-    const fn_res = local_eval(fn_eval_str);
+    const fn_res = eval(fn_eval_str);
```
Comment (lambdageek) [inline src/mono/wasm/runtime/src/mono/debug.ts:162]: You could perhaps do something like this:

```ts
const fn_eval_str = `function (proxy) { var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]); }`;
const fn_defn = local_eval (fn_eval_str);
const fn_res = fn_defn (proxy);
```

Comment (lambdageek) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: Or maybe we should just use the `Function` constructor.

```ts
const fn_defn = new Function ('proxy', `var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]);`);
const fn_res = fn_defn (proxy);
```

Comment (lambdageek) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: Sorry to code golf this.  I'm confused why we need `eval` at all.

Why isn't it just:
```ts
  const fn = request.functionDeclaration;
  const fn_res = fn.call (proxy, ...fn_args);
```

Comment (radical) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: >   const fn = request.functionDeclaration;

`request.functionDeclaration` will be a string (eg. https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L26 and https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L79). Would this work? If so, then we could do what you are suggesting.

Comment (thaystg) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: It doesn't compile, I'm testing other approach based on @lambdageek suggestions.

Comment (lambdageek) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: > > const fn = request.functionDeclaration;
> 
> `request.functionDeclaration` will be a string (eg. https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L26 and https://github.com/dotnet/runtime/blob/main/src/mono/wasm/debugger/DebuggerTestSuite/CallFunctionOnTests.cs#L79). Would this work? If so, then we could do what you are suggesting.

Ah ok.  So we really do need to eval it.

I think the `new Function(...)` approach should work.

Comment (pavelsavara) [reply src/mono/wasm/runtime/src/mono/debug.ts:162]: Sorry everybody, I was not aware that debugger tests are not running on CI :(

File: src/mono/wasm/runtime/src/mono/debug.ts [hunk 2]
```diff
@@ -157,10 +157,11 @@ export function mono_wasm_call_function_on(request: CallRequest) {
     }
 
     const fn_args = request.arguments != undefined ? request.arguments.map(a => JSON.stringify(a.value)) : [];
-    const fn_eval_str = `var fn = ${request.functionDeclaration}; fn.call (proxy, ...[${fn_args}]);`;
 
-    const local_eval = eval; // https://rollupjs.org/guide/en/#avoiding-eval
-    const fn_res = local_eval(fn_eval_str);
+    const fn_body_template = `var fn = ${request.functionDeclaration}; return fn.call (proxy, ...[${fn_args}]);`;
```
Comment (kg) [inline src/mono/wasm/runtime/src/mono/debug.ts:161]: just use fn.apply, there's no reason to do .call and then spread in the arglist

**Review by lambdageek — APPROVED** [review]

**Pull request merged**